### ISSUE TITLE
Missing dependencies in fedora

### ISSUE THREAD

**Issue by jbtrystram** [drogue-iot/drg#85]
Crates that need fedora packaging
- base64
- rcgen
- json_merge_values

**Comment by decathorpe** [comment]
I've made progress with some of those missing pieces:

- json_merge_values is reviewed and ready for importing to Fedora
- rand_isaac is updated to version 0.3.0
- tungstenite is updated to version 0.14.0

As for the RustCrypto packages, I already have the whole dependency tree of rsa 0.5.0 built successfully in a COPR for a different reason:
https://copr.fedorainfracloud.org/coprs/decathorpe/deno/

However, the complete dependency tree of missing packages and updates (that does not only include minimal build dependencies, but actually *all* dependencies) is significantly larger than just the 4 packages you listed (sorry for the messy drawing, I only have this as a hand-drawn graph to help me think):

![rsa-0 5 0-dep-tree](https://user-images.githubusercontent.com/12836693/145565439-cf50c1d0-bad1-47b1-a9e1-25fa3b58f91b.png)

I have all the necessary packages basically ready for review or update, but it will take time and somebody to actually review the packages (and probably obsoleting some old package reviews that have been stale for months, and are not for the most recent versions any longer).

**Comment by jbtrystram** [comment]
yeah I did not go too far in that rabbit hole :)  Thanks for doing that work!   I established the list several months ago, so I am not surprised it's not up to date.
Let me know if I can help in the packaging effort

**Comment by decathorpe** [comment]
I'll start submitting some of the missing packages for review, and if you could help with reviewing them, that would help a lot.

**Comment by jbtrystram** [comment]
Sure, i'll do that ! I was not aware than anyone from the packager group could review packages. I thought that I needed to be promoted to another group to be able to review packages. TIL !

**Comment by jbtrystram** [comment]
@decathorpe after a bit of looking around I found that there are new major versions of PKCS1 (0.3.1), PKCS8 (0.8.0) and DER (0.5.1) released. 
From this issue https://github.com/RustCrypto/RSA/issues/120, maybe it won't be too long before the RSA crate update as well, so it may be worth waiting a bit for RSA to catch up..
What do you think ?

**Comment by decathorpe** [comment]
I don't think waiting is a good idea. I won't be able to update the whole RustCrypto stack immediately anyway since there's a large number of packages depending on it. I'd rather bring in the packages we need now, and I can always update them later when everything is ready and can be done in one step. Hope that makes sense.

**Comment by decathorpe** [comment]
I've submitted three more packages for review:

https://bugzilla.redhat.com/show_bug.cgi?id=2031925
https://bugzilla.redhat.com/show_bug.cgi?id=2031927
https://bugzilla.redhat.com/show_bug.cgi?id=2032563

I'm also waiting for a response on the pre-existing der / const-oid / spki / pkcs8 review requests before obsoleting them with tickets for newer versions.

With those out of the way, that leaves only a handful more of missing packages (salsa20, scrypt, pbkdf2, pkcs5, pkcs1) which I will submit for review once their dependencies are available from Fedora repositories.

**Comment by decathorpe** [comment]
I've submitted the remaining missing packages for rsa 0.5.0 for review:

https://bugzilla.redhat.com/show_bug.cgi?id=2034211
https://bugzilla.redhat.com/show_bug.cgi?id=2034212
https://bugzilla.redhat.com/show_bug.cgi?id=2034213
https://bugzilla.redhat.com/show_bug.cgi?id=2034218
https://bugzilla.redhat.com/show_bug.cgi?id=2034224
https://bugzilla.redhat.com/show_bug.cgi?id=2034225
https://bugzilla.redhat.com/show_bug.cgi?id=2034226
https://bugzilla.redhat.com/show_bug.cgi?id=2034228
https://bugzilla.redhat.com/show_bug.cgi?id=2034230

**Comment by decathorpe** [comment]
The dependencies for rsa 0.5.0 are now in Fedora, though pbrobinson has broken his packages, so updating them will take some time. I'll probably also be able to update webpki to 0.22 soon (as soon as reqwest will move to the latest trust-dns versions).

**Comment by jbtrystram** [comment]
Thanks a lot !

**Comment by decathorpe** [comment]
Can you remind me of which updates / packages are still missing?

I've been churning through the rust-sig backlog for the past few days, just let me know which packages I still need to prioritize.
